File: jumping_on_clouds.py
#!/usr/bin/python3

# Complete the jumpingOnClouds function below.
def jumpingOnClouds(c):
    # c = [0,0,1,0,0,1,0]

    jumps = 0
    current_position = 0
    while current_position < len(c) - 1:
        #                                      |
        # protect from going out of bounds ie. v
        #                             [0,0,0,1,0,0]
        if current_position + 2 <= len(c) - 1 and c[current_position + 2] == 0:
            current_position += 2
        else:
            current_position += 1
        jumps += 1
    print(jumps)
    return jumps


jumpingOnClouds([0, 1, 0, 0, 0, 1, 0])
jumpingOnClouds([0, 0, 1, 0, 0, 1, 0])
jumpingOnClouds([0, 0, 0, 0, 1, 0])

File: jumping_on_clouds.go
package main

import "fmt"

func jumpingOnClouds(c []int) (jumps int) {
	currentPosition := 0
	for currentPosition < len(c)-1 {
		if currentPosition+2 <= len(c) && c[currentPosition+2] == 0 {
			fmt.Println("jumping", 2)
			currentPosition += 2
		} else {
			fmt.Println("jumping", 1)
			currentPosition += 1
		}
		jumps += 1
	}
	fmt.Println(jumps)
	return jumps
}

func main() {
	ar1 := []int{0, 1, 0, 0, 0, 1, 0}
	jumpingOnClouds(ar1)

	ar2 := []int{0, 0, 1, 0, 0, 1, 0}
	jumpingOnClouds(ar2)

	ar3 := []int{0, 0, 0, 0, 1, 0}
	jumpingOnClouds(ar3)
}
